### PR TITLE
Allow users to override the default template's layout file

### PR DESCRIPTION
**Pull request by kdelmonte**
This addresses both my concerns and the author's in #6.

After reading [these instructions](http://usejsdoc.org/about-configuring-default-template.html#overriding-the-default-template-s-layout-file) from the jsdoc team and looking at https://github.com/jsdoc3/jsdoc/issues/480, I feel confident that this change is the way to go.

### CODE REVIEW

File: src/gulp-jsdoc.js [hunk 1]
```diff
@@ -29,6 +29,12 @@ var debug = require('debug')('gulp-jsdoc3');
 export function jsdoc(config, done) {
     var files = [];
     var jsdocConfig = config || require('./jsdocConfig.json');
+    var hasUserSpecifiedTemplate = false;
+    var args;
+
+    if(jsdocConfig.templates && jsdocConfig.templates.default && jsdocConfig.templates.default.layoutFile){
```
Comment (mlucool) [inline src/gulp-jsdoc.js]: Move below, no need to figure this out ahead of time.

File: src/gulp-jsdoc.js [hunk 2]
```diff
@@ -62,8 +68,15 @@ export function jsdoc(config, done) {
                 const spawn = require('child_process').spawn,
                     cmd = require.resolve('jsdoc/jsdoc.js'), // Needed to handle npm3 - find the binary anywhere
                     inkdocstrap = path.dirname(require.resolve('ink-docstrap'));
+                
+                args = ['-c', tmpobj.name];
+                
                 // Config + ink-docstrap
-                const args = ['-c', tmpobj.name, '-t', inkdocstrap].concat(files);
+                if(!hasUserSpecifiedTemplate){
```
Comment (mlucool) [inline src/gulp-jsdoc.js]: Move `if(jsdocConfig.templates && jsdocConfig.templates.default && jsdocConfig.templates.default.layoutFile)` from above here

File: src/gulp-jsdoc.js [hunk 3]
```diff
@@ -29,6 +29,12 @@ var debug = require('debug')('gulp-jsdoc3');
 export function jsdoc(config, done) {
     var files = [];
     var jsdocConfig = config || require('./jsdocConfig.json');
+    var hasUserSpecifiedTemplate = false;
+    var args;
```
Comment (mlucool) [inline src/gulp-jsdoc.js]: Why does args need to be moved up?

File: README.md [hunk 4]
```diff
@@ -11,13 +11,13 @@ Install `gulp-jsdoc` as a development dependency:
 npm install --save-dev gulp-jsdoc3
 ```
 
-Then, use it where config is the only way to pass in jsdoc options. All CLI options are can be specified here,
-the only exception is ink-docstrap is bundled here and used for templating.
+The config is the only way to pass in jsdoc options. All CLI options are can be specified here.
```
Comment (mlucool) [inline README.md]: Mention that ink-docstrap is the default

File: README.md [hunk 5]
```diff
@@ -11,13 +11,13 @@ Install `gulp-jsdoc` as a development dependency:
...
 
 ```javascript
 var jsdoc = require('gulp-jsdoc3');
 
 gulp.task('doc', function (cb) {
+    var config = require('./jsdoc.json');
```
Comment (mlucool) [inline README.md]: Config isn't needed, it should actually be passed in with undefined to use the default implementation. My oversight

**Comment by mlucool** [comment]
:+1: Pretty good, just a few changes based on a quick glance at the code.

File: src/gulp-jsdoc.js (hunk 3 above)
Comment (kdelmonte) [inline src/gulp-jsdoc.js]: ESLint was complaining.

**Comment by kdelmonte** [comment]
The changes that you requested are done. Once you approve, I'll squash the commits to a single one so you can merge.

**Comment by vbardales** [comment]
:+1:

**Comment by mlucool** [comment]
Seems the build is failing with this comment. Can you check why?

There is also a dev-dependency issue failing unrelated to this comment:
$ gulp lint
[23:01:48] Using gulpfile ~/gulp-jsdoc3/gulpfile.js
[23:01:48] Starting 'lint'...
TypeError: Cannot read property 'visitClass' of undefined
    at monkeypatch (/gulp-jsdoc3/node_modules/babel-eslint/index.js:220:40)

So without linting I can't test this, but I am pretty confident that you can move that added var down as a const. What is the error you see when you do?

:+1: for what you did for allowing a done callback only

**Comment by kdelmonte** [comment]
"vars-on-top": 2 fromt he .eslintrc  is what triggered the error. `args` is no longer a const so I cannot move it down... Let me know... Good night.

**Comment by mlucool** [comment]
Use 'let' instead of const - we still want block scoping. I should change all my vars to let also

**Comment by kdelmonte** [comment]
Auf Wiedersehen vars.

**Comment by mlucool** [comment]
:+1: this looks great. Really appreciate the iterations! Please squash and I'll merge. The release won't be for another 2 weeks due to travel conflicts.

**Comment by kdelmonte** [comment]
Done. Thank you.

**Comment by mlucool** [comment]
Thanks again!

**Pull request merged**